Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -599,10 +610,11 @@ class CombinatorialKalmanFilter {
                          : TrackStatePropMask::All);
 
           // Add measurement/outlier track state to the multitrajectory
+          auto source_link_it = std::next(lower_it, index);
```

Comment:
What does this do? Advance the iterator by `index`?

Reply:
Yes. The `index` is the index of the measurement on a detector module, and we want to access the measurement which has been selected.